test/session-factory.ts

```diff
@@ -0,0 +1,237 @@
+/*!
+ * Copyright 2024 Google LLC. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {Database, Session, SessionPool} from '../src';
+import {SessionFactory} from '../src/session-factory';
+import * as sinon from 'sinon';
+import * as assert from 'assert';
+import {MultiplexedSession} from '../src/multiplexed-session';
+import {util} from '@google-cloud/common';
+import * as db from '../src/database';
+import {FakeTransaction} from './session-pool';
+import {ReleaseError} from '../src/session-pool';
+
+describe('SessionFactory', () => {
+  process.env.GOOGLE_CLOUD_SPANNER_MULTIPLEXED_SESSIONS = 'true';
```

Comment:
Can we initialize this in `before` block and clear in `after` block

Reply:
we need to keep this in after block so that others test which will be running after this test don't fail on because of the change in env configuration